### DISCUSSION TITLE
Error while building application in cqube

### DISCUSSION THREAD

**Discussion by aartiadkhale** [Sunbird-cQube/community#865] [General]
Error building dashboard_ms - code: 1, message: The command '/bin/sh -c npm run build --prod' returned a non-zero code: 1
91mnpm notice created a lockfile as package-lock.json. You should commit this file.', '\\x1b[0m', '\\x1b[91mnpm

**Comment by bommalla** [comment]
@aartiadkhale 
Thanks for reaching out. Which version of cQube are you currently installed? And could you please provide details about the exact issue you encountered

**Reply by aartiadkhale** [reply]
Version - cQube v5.0
I attached error logs 
[build-error.docx](https://github.com/Sunbird-cQube/community/files/14368657/build-error.docx)

**Reply by bommalla** [reply]
@aartiadkhale 

Could you please post the steps you executed to rebuild the dashboard ?

**Reply by archanamakude** [reply]
It's cQube version 5.0.3 for state vsk

**Comment by aartiadkhale** [comment]
Building docker image and Docker files contains
FROM node:14.21.2 as build
WORKDIR /app
COPY package.json .
RUN npm install --legacy-peer-deps
COPY . .
RUN npm run build --prod

**Reply by bommalla** [reply]
@aartiadkhale 

Please confirm me which version of cQube are you installed v5.0.0 or v5.0.3 ?

**Reply by aartiadkhale** [reply]
v5.0.3

**Comment by bommalla** [comment]
@aartiadkhale 

Please find the below rebuilding steps for dashboard-ms

Dashboard redeployment steps:

sudo docker stop dashboard_app
sudo docker rm dashboard_app
sudo docker rmi dashboard_ms:1

Go inside cQube-devops/microservices/dashboard-ms then run the below command
sudo docker build -t dashboard_ms:1 .

After that navigate the below mentioned path 
cd cQube-devops/ansible run the below command
sudo docker-compose -f docker-compose.yml up -d

**Reply by aartiadkhale** [reply]
After following above steps getting this error

Pulling dashboard_app (dashboard_ms:1)...
ERROR: pull access denied for dashboard_ms, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

**Comment by bommalla** [comment]
@aartiadkhale 

Just run the below command and sens the screenshot 

sudo docker ps -a

**Reply by aartiadkhale** [reply]
docker image not created 
![docker-image-not-created](https://github.com/Sunbird-cQube/community/assets/148222875/011aee33-6e12-49b9-acb2-b32ba043f5f5)

**Reply by bommalla** [reply]
@aartiadkhale 

Okey,Please follow the below steps and re install the cQube 

1.  Please navigate the below mentioned path

cd cqube-devops/microserices/dashboard-ms

sudo vim Dockerfile

Add the package-lock.json file into to the dockerfile 

2. After that remove the all existing containers by using below commands
sudo docker stop $(sudo docker ps -aq)
sudo docker rm $(sudo docker ps -aq)
sudo docker rmi $(sudo docker images -q)
sudo docker volume prune -y
sudo docker volume prune
sudo docker volume rm adapter_volume monitoring_grafana_data monitoring_prometheus_data python_executable_env
sudo docker network prune

sudo rm -rf /opt/cqube
sudo rm -rf /opt/containerd

3. Then you can reinstall the cQube by using below command
sudo ./install.sh

**Reply by aartiadkhale** [reply]
@bommalla  Thank you so much.
Adding package-lock.json in Docker file but still not able to create image please check attached error. 
[build-error-2.docx](https://github.com/Sunbird-cQube/community/files/14371927/build-error-2.docx)

**Comment by bommalla** [comment]
@aartiadkhale 

Let me check from end then I'll get back to you

**Comment by bommalla** [comment]
@aartiadkhale 

I'm pleased to inform you that we have successfully resolved the issue regarding the automatic update of dashboard-ms packages. The problem has been fixed, and everything is now functioning as expected.

I'd like to inform you that we've just merged the code. Therefore, please clone the repository and reinstall it to ensure that the changes are correctly reflected.

Thank you for your patience and cooperation throughout this process. Should you encounter any further issues or have any questions, please don't hesitate to reach out.

**Reply by aartiadkhale** [reply]
Thank you for your quick response, we merged code and if any issue let you know.

**Comment by bommalla** [comment]
@aartiadkhale 

Thanks for the update.
